lang/java/avro/src/test/java/org/apache/avro/TestSchemas.java

```diff
@@ -0,0 +1,139 @@
+      .createUnion(list(INT_SCHEMA, LONG_SCHEMA, FLOAT_SCHEMA, DOUBLE_SCHEMA));
+
+  // Non recursive records:
+  static final Schema EMPTY_RECORD1 = Schema.createRecord("Record1", null, null, false);
+  static final Schema EMPTY_RECORD2 = Schema.createRecord("Record2", null, null, false);
+  static final Schema A_INT_RECORD1 = Schema.createRecord("Record1", null, null, false);
+  static final Schema A_LONG_RECORD1 = Schema.createRecord("Record1", null, null, false);
+  static final Schema A_INT_B_INT_RECORD1 = Schema.createRecord("Record1", null, null, false);
+  static final Schema A_DINT_RECORD1 = // DTYPE means TYPE with default value
+      Schema.createRecord("Record1", null, null, false);
+  static final Schema A_INT_B_DINT_RECORD1 = Schema.createRecord("Record1", null, null, false);
+  static final Schema A_DINT_B_DINT_RECORD1 = Schema.createRecord("Record1", null, null, false);
+
+  static final Schema FIXED_4_BYTES = Schema.createFixed("Fixed", null, null, 4);
+  static final Schema FIXED_8_BYTES = Schema.createFixed("Fixed", null, null, 8);
+
+  static {
+    EMPTY_RECORD1.setFields(Collections.<Field>emptyList());
+    EMPTY_RECORD2.setFields(Collections.<Field>emptyList());
+    A_INT_RECORD1.setFields(list(new Field("a", INT_SCHEMA, null, null)));
+    A_LONG_RECORD1.setFields(list(new Field("a", LONG_SCHEMA, null, null)));
+    A_INT_B_INT_RECORD1.setFields(
+        list(new Field("a", INT_SCHEMA, null, null), new Field("b", INT_SCHEMA, null, null)));
+    A_DINT_RECORD1.setFields(list(new Field("a", INT_SCHEMA, null, 0)));
+    A_INT_B_DINT_RECORD1.setFields(
+        list(new Field("a", INT_SCHEMA, null, null), new Field("b", INT_SCHEMA, null, 0)));
+    A_DINT_B_DINT_RECORD1
+        .setFields(list(new Field("a", INT_SCHEMA, null, 0), new Field("b", INT_SCHEMA, null, 0)));
+  }
+
+  // Recursive records
+  static final Schema INT_LIST_RECORD = Schema.createRecord("List", null, null, false);
+  static final Schema LONG_LIST_RECORD = Schema.createRecord("List", null, null, false);
+  static {
+    INT_LIST_RECORD.setFields(list(new Field("head", INT_SCHEMA, null, null),
+        new Field("tail", INT_LIST_RECORD, null, null)));
+    LONG_LIST_RECORD.setFields(list(new Field("head", LONG_SCHEMA, null, null),
+        new Field("tail", LONG_LIST_RECORD, null, null)));
+  }
+
+  // -----------------------------------------------------------------------------------------------
+
+  /** Reader/writer schema pair. */
+  static final class ReaderWriter {
+    private final Schema mReader;
+    private final Schema mWriter;
+
+    public ReaderWriter(final Schema reader, final Schema writer) {
+      mReader = reader;
+      mWriter = writer;
+    }
+
+    public Schema getReader() {
+      return mReader;
+    }
+
+    public Schema getWriter() {
+      return mWriter;
+    }
+  }
+
+  /** Borrowed from the Guava library. */
+  static <E> ArrayList<E> list(E... elements) {
```

Comment:
Not sure that we need this at all, since it does the same as Arrays.asList(), though better not change this in this PR, since it is already non-trivial.

Reply:
It was there before me :-)
I also think it is better to leave it for now
Could we make a "cleaning JIRA", where we also could tidy up the formatting of SchemaCompatibility/SchemaValidator related classes?

Reply:
@epkanol yes, I think a separate Jira is the best option to cleanup the tests related the schema compatibility/schema validation.